src/rules/selector-pseudo-class-whitelist/index.js

```diff
@@ -0,0 +1,58 @@
+import {
+  isStandardSyntaxSelector,
+  matchesStringOrRegExp,
+  parseSelector,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+import { isString } from "lodash"
+import { vendor } from "postcss"
+
+export const ruleName = "selector-pseudo-class-whitelist"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: (selector) => `Unexpected pseudo-class "${selector}"`,
+})
+
+function rule(whitelist) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: whitelist,
+      possible: [isString],
+    })
+    if (!validOptions) { return }
+
+    root.walkRules(rule => {
+      const { selector } = rule
+
+      if (!isStandardSyntaxSelector(selector)) { return }
+      if (selector.indexOf(":") === -1) { return }
+
+      parseSelector(selector, result, rule, selectorTree => {
+        selectorTree.walkPseudos(pseudoNode => {
+          const pseudo = pseudoNode.value
+
+          // Ignore pseudo-elements
+          if (pseudo.indexOf("::") !== -1) { return }
```

Comment:
Is there any reason not to change this line to 

```js
if (pseudo.slice(0, 2) !== "::") { return }
```

which would not search the whole string when only the beginning matters?